src/execution/utils.ts

```diff
@@ -397,7 +404,30 @@ function getTransactionParams(transaction: Transaction) {
     eip7702InitSignature,
     gasLimit,
     settlementLayers,
+    sourceAssets,
+    feeAsset,
+  }
+}
+
+function getTokenRequests(
+  sourceChains: Chain[],
+  targetChain: Chain,
+  initialTokenRequests: TokenRequest[] | undefined,
+) {
+  // Across requires passing some value to repay the solvers
+  const defaultTokenRequest = {
+    address: zeroAddress,
+    amount: 1n,
   }
+  const isSameChain =
+    sourceChains.length === 1 && sourceChains[0].id === targetChain.id
```

Comment:
Btw there's no proper way for an SDK to know what settlement layer we will be using for an intent (unless `settlementLayers` is passed)